ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh

```diff
@@ -0,0 +1,475 @@
+      if [ -f "${SHARED_DIR}/proxy-conf.sh" ]; then
+          # shellcheck source=/dev/null
+          source "${SHARED_DIR}/proxy-conf.sh"
+      fi
+
+      export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+      mirror_support_tools
+  fi
+}
+
+wait_for_mcp() {
+  timeout=${1}
+  # Wait until MCO starts applying new machine config to nodes
+  log "Waiting for all MachineConfigPools to start updating..."
+  time oc wait mcp --all --for='condition=UPDATING=True' --timeout=300s &>/dev/null
+
+  log "Waiting for all MachineConfigPools to finish updating..."
+  timeout "${timeout}" bash <<EOT
+    until
+      oc wait mcp --all --for='condition=UPDATED=True' --timeout=10s 2>/dev/null && \
+      oc wait mcp --all --for='condition=UPDATING=False' --timeout=10s 2>/dev/null && \
+      oc wait mcp --all --for='condition=DEGRADED=False' --timeout=10s;
+    do
+      sleep 10
+    done
+EOT
+  log "All MachineConfigPools to finished updating"
+}
+
+wait_for_co() {
+  timeout=${1}
+  log "Waiting for all ClusterOperators to update..."
+  timeout "${timeout}" bash <<EOT
+  until
+    oc wait co --all --for='condition=AVAILABLE=True' --timeout=10s &>/dev/null && \
+    oc wait co --all --for='condition=PROGRESSING=False' --timeout=10s &>/dev/null && \
+    oc wait co --all --for='condition=DEGRADED=False' --timeout=10s &>/dev/null;
+  do
+    sleep 10
+  done
+EOT
+  log "All ClusterOperators finished updating"
+}
+
+wait_for_cno_cleanup() {
+  timeout=${1}
+  log "Waiting for CNO to clean migration status..."
+  timeout "${timeout}" bash <<EOT
+  until
+    ! oc get network -o yaml | grep migration > /dev/null
+  do
+    echo "migration field is not cleaned by CNO"
+    sleep 10
+  done
+EOT
+  log "CNO cleaned migration status"
+}
+
+wait_for_final_mc() {
+  timeout=${1}
+  log "Waiting for final MachineConfigs..."
+  script=$(sed -e "s,%%data%%,${encoded_set_mtu},g" <<'EOT'
+  while [ "${is_final:-0}" = "0" ]; do
+    is_final=1
+    for pool in worker master; do
+      mc_name=$(oc get mc -o custom-columns=:metadata.name --sort-by=.metadata.creationTimestamp | grep rendered-$pool | tail -1)
+      mc=$(oc get mc $mc_name -o yaml || true)
+      [ -z "$mc" ] && is_final=0 && break
+      echo $mc | grep -q "mtu-migration.sh" && is_final=0 && break
+      echo $mc | grep -q "%%data%%" || (is_final=0 && break)
+    done
+    [ "$is_final" = "1" ] || sleep 10
+  done
+EOT
+)
+  timeout "${timeout}" bash <<SCRIPT
+  $script
+SCRIPT
+  log "MachineConfigs are at their final state"
+}
+
+wait_for_final_mc_controller_config() {
+  timeout=${1}
+  log "Waiting for final Machine Controller Config..."
+  timeout "${timeout}" bash <<EOT
+  until
+    ! oc get controllerconfigs machine-config-controller -o yaml | grep mtuMigration > /dev/null
+  do
+    echo "migration field is not cleaned by MCO"
+    sleep 10
+  done
+EOT
+  log "Machine Controller Config is at its final state"
+}
+
+check_no_api_alerts() {
```

Comment:
Don't we have a test which ensures no important alerts are firing during the test?

Reply:
Hmm I didn't know that. Which period does it cover? The whole test run or only while e2e tests are running?